spring-kafka/src/main/java/org/springframework/kafka/config/KafkaListenerEndpointRegistry.java

```diff
@@ -270,7 +271,10 @@ public void stop() {
 	public void stop(Runnable callback) {
 		Collection<MessageListenerContainer> listenerContainersToStop = getListenerContainers();
 		if (listenerContainersToStop.size() > 0) {
-			AggregatingCallback aggregatingCallback = new AggregatingCallback(listenerContainersToStop.size(), callback);
+			AggregatingCallback aggregatingCallback = new AggregatingCallback(listenerContainersToStop.size(), () -> {
+				this.running = false;
```

Comment:
Well, you still have this logic inside a callback call and that's exactly what you try to justify is wrong anyway.
My point is to have a logic like this:
```
public void stop(Runnable callback) {
this.running = false;
...
```

So, this way we are stopped and it the state is changes exactly before we call a callback.

Reply:
Ok, will change.